x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/DeleteEngineAction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class DeleteEngineAction extends ActionType<AcknowledgedResponse> {
+
+    public static final DeleteEngineAction INSTANCE = new DeleteEngineAction();
+    public static final String NAME = "indices:admin/engine/delete";
+
+    private DeleteEngineAction() {
+        super(NAME, AcknowledgedResponse::readFrom);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
+
+        public static final IndicesOptions DEFAULT_INDICES_OPTIONS = IndicesOptions.strictExpandOpen();
+
+        private String[] names;
+        private final IndicesOptions indicesOptions = DEFAULT_INDICES_OPTIONS;
+
+        private final String engineId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.engineId = in.readString();
+            names = new String[] { Engine.getEngineAliasName(this.engineId) };
```

Comment:
Should we move `getEngineAliasName` to the `Engine` class?

Reply:
I'm not sure exactly what you're thinking, since we don't have a constructed Engine object yet and it has to be called this way?

Reply:
Oh - you're right. It reminded me of something of the past PR 👍